testsuite/cluster-test/src/health/fullnode_check.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::prometheus::Prometheus;
+use crate::util::unix_timestamp_now;
+use crate::{
+    cluster::Cluster,
+    health::{HealthCheck, HealthCheckContext, ValidatorEvent},
+};
+use anyhow::{format_err, Result};
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::collections::HashMap;
+use std::time::Duration;
+
+const APP_LIBRA_VALIDATOR: &str = "libra-validator";
+const APP_LIBRA_FULLNODE: &str = "libra-fullnode";
+static VAL_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"val-(\d+)").unwrap());
+static FULLNODE_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"fn-(\d+)").unwrap());
+
+pub struct FullNodeHealthCheck {
+    cluster: Cluster,
+    prometheus: Option<Prometheus>,
```

Comment:
FullNodeHealthCheck does not work without prometheus at all right? In this case I think we want this to be non-optional, and just skip initializing FullNodeHealthCheck if there is no prometheus

Reply:
u are correct, this will be more clear instead of option